test/auth_cluster/test_maintenance_port.py

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright (C) 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+from cassandra.auth import PlainTextAuthProvider
+from cassandra.cluster import Cluster, NoHostAvailable
+from cassandra import Unauthorized
+from cassandra.connection import UnixSocketEndPoint
+from test.pylib.manager_client import ManagerClient
+
+import pytest
+
+@pytest.mark.asyncio
+async def test_maintenance_port(manager: ManagerClient):
+    """
+    Test that when connecting to the maintenance port, the user has superuser permissions,
+    even if the authentication is enabled on the regular port.
+    """
+    socket = "/tmp/test_socket"
+
+    server = await manager.server_add(config={
+        "maintenance_socket": socket,
+        "authenticator": "PasswordAuthenticator",
+        "authorizer": "CassandraAuthorizer",
+    })
+
+    try:
+        cluster = Cluster([server.ip_addr])
+        session = cluster.connect()
+        pytest.fail("Should have failed")
```

Comment:
More pythonic would be to put this into `else` branch.
The message should be more descriptive, "Should have failed" says nothing more than using `pytest.fail("")`.